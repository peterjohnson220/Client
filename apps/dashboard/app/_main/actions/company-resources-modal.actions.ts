import { Action } from '@ngrx/store';


export const OPENING_NEW_FOLDER_MODAL = '[Company Resources/Company Resources Page Modal] Opening New Folder Modal';
export const CLOSING_NEW_FOLDER_MODAL = '[Company Resources/Company Resources Page Modal] Closing New Folder Modal';


export class OpeningNewFolderModal implements Action {
  readonly type = OPENING_NEW_FOLDER_MODAL;
}

export class ClosingNewFolderModal implements Action {
  readonly type = CLOSING_NEW_FOLDER_MODAL;
}


export type Actions
  = OpeningNewFolderModal
  | ClosingNewFolderModal;
