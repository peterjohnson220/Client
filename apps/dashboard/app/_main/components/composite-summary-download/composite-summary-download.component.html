<!-- ngIf syntax here is a hack to make the contents of the file$ observable available as a local template variable -->
<div class="w-75 mx-auto mt-4 text-white text-center" *ngIf="(file$ | async) || {} as file">
  <h1>Downloading file...</h1>
  <div *ngIf="file.error || errorText; else downloading">
    <h2 class="text-danger" *ngIf="file.errorText">Error: {{ file.errorText }}</h2>
    <h2 class="text-danger" *ngIf="errorText">Error: {{ errorText }}</h2>
  </div>
  <ng-template #downloading>
    <div>
      <h2 *ngIf="file.fileName; else inProgress">Download Complete.</h2>
      <ng-template #inProgress>
        <h2>Please stand by.</h2>
      </ng-template>
      <ngb-progressbar class="d-block mt-4" [showValue]="true" type="success" [value]="file.progress || 0"></ngb-progressbar>
    </div>
  </ng-template>
</div>
