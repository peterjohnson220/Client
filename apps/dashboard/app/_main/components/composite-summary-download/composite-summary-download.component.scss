p {
  color: white;
  font-size: 200%;
}

div {
  width: 100%;
  text-align: center;
  margin-top: 30px;
}
