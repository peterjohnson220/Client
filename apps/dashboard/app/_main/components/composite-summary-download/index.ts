export * from './composite-summary-download.component';
