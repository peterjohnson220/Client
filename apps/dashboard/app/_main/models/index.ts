export * from './feature.model';
export * from './feature-type';
export * from './feature-types';
export * from './tile.model';
export * from './tile-preview-base.model';
export * from './tile-preview-basic-list-tile';
export * from './tile-preview-chart.model';
export * from './tile-preview-chart-with-calendar.model';
export * from './tile-preview-chart-data.model';
export * from './tile-preview-chart-with-calendar-data.model';
export * from './tile-preview-chart-with-list-model';
export * from './tile-preview-chart-type';
export * from './tile-preview-chart-types';
export * from './tile-preview-icon.model';
export * from './tile-preview-list.model';
export * from './tile-preview-place-holder.model';
export * from './tile-preview-type';
export * from './tile-preview-types';
export * from './tile-type';
export * from './tile-types';
export * from './timeline-activity.model';
export * from './timeline-activity-filter.model';
