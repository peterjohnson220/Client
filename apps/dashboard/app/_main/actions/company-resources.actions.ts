import { HttpErrorResponse } from '@angular/common/http';
import { Action } from '@ngrx/store';
import { CompanyResourcePost, CompanyResources } from '../models';

export const GETTING_COMPANY_RESOURCES = '[Company Resources/Company Resources Page] Getting Company Resources';
export const GETTING_COMPANY_RESOURCES_SUCCESS = '[Company Resources/Company Resources Page] Getting Company Resources Success';
export const GETTING_COMPANY_RESOURCES_ERROR = '[Company Resources/Company Resources Page] Getting Company Resources Error';

export const ADDING_COMPANY_RESOURCE = '[Company Resources/Company Resources Page] Adding Company Resource';
export const ADDING_COMPANY_RESOURCE_SUCCESS = '[Company Resources/Company Resources Page] Adding Company Resource Success';
export const ADDING_COMPANY_RESOURCE_ERROR = '[Company Resources/Company Resources Page] Adding Company Resource Error';

export const DELETING_COMPANY_RESOURCE = '[Company Resources/Company Resources Page] Deleting Company Resource';
export const DELETING_COMPANY_RESOURCE_SUCCESS = '[Company Resources/Company Resources Page] Deleting Company Resource Success';
export const DELETING_COMPANY_RESOURCE_ERROR = '[Company Resources/Company Resources Page] Deleting Company Resource Error';

export class GettingCompanyResources implements Action {
  readonly type = GETTING_COMPANY_RESOURCES;
}

export class GettingCompanyResourcesSuccess implements Action {
  readonly type = GETTING_COMPANY_RESOURCES_SUCCESS;

  constructor(public payload: CompanyResources) {}
}

export class GettingCompanyResourcesError implements Action {
  readonly type = GETTING_COMPANY_RESOURCES_ERROR;

  constructor(public error: HttpErrorResponse) {}
}

export class AddingCompanyResource implements Action {
  readonly type = ADDING_COMPANY_RESOURCE;

  constructor(public payload: CompanyResourcePost) {}
}

export class AddingCompanyResourceSuccess implements Action {
  readonly type = ADDING_COMPANY_RESOURCE_SUCCESS;

  constructor(public payload) {}
}

export class AddingCompanyResourceError implements Action {
  readonly type = ADDING_COMPANY_RESOURCE_ERROR;

  constructor(public error: HttpErrorResponse) {}
}

export class DeletingCompanyResource implements Action {
  readonly type = DELETING_COMPANY_RESOURCE;

  constructor(public payload: number) {}
}

export class DeletingCompanyResourceSuccess implements Action {
  readonly type = DELETING_COMPANY_RESOURCE_SUCCESS;

  constructor(public payload) {}
}

export class DeletingCompanyResourceError implements Action {
  readonly type = DELETING_COMPANY_RESOURCE_ERROR;

  constructor(public error: HttpErrorResponse) {}
}

export type Actions
  = GettingCompanyResources
  | GettingCompanyResourcesSuccess
  | GettingCompanyResourcesError
  | AddingCompanyResource
  | AddingCompanyResourceSuccess
  | AddingCompanyResourceError
  | DeletingCompanyResource
  | DeletingCompanyResourceSuccess
  | DeletingCompanyResourceError;
