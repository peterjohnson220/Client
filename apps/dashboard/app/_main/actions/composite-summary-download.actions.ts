import { Action } from '@ngrx/store';
import { Params } from '@angular/router';

import { CompositeSummaryDownloadRequest } from 'libs/models/dashboard';

export const FETCH_AUTH_JWT = '[Dashboard/Composite Summary Download] Fetch Authentication JWT';
export const COMPOSITE_SUMMARY_DOWNLOAD = '[Dashboard/Composite Summary Download] Composite Summary Download';

export class FetchAuthJWT implements Action {
  readonly type = FETCH_AUTH_JWT;

  constructor(public payload: Params) {}
}

export class CompositeSummaryDownload implements Action {
  readonly type = COMPOSITE_SUMMARY_DOWNLOAD;

  constructor(public payload: CompositeSummaryDownloadRequest) {}
}

export type Actions =
  FetchAuthJWT |
  CompositeSummaryDownload;
