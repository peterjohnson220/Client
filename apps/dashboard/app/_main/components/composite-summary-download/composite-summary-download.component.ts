import { Component, OnDestroy, OnInit } from '@angular/core';
import { ActivatedRoute, ParamMap } from '@angular/router';
import { select, Store } from '@ngrx/store';
import { Observable, Subscription } from 'rxjs';

import { RegexStrings } from 'libs/constants/regex-strings';
import * as fromFileReducer from 'libs/features/file-download/reducers/file-download.reducer';
import { FileModel } from 'libs/models/file';
import { CompositeSummaryDownload } from '../../actions/composite-summary-download.actions';
import { FileType } from 'libs/models/dashboard';

@Component({
  selector: 'pf-composite-summary-download',
  templateUrl: './composite-summary-download.component.html',
  styleUrls: ['./composite-summary-download.component.scss']
})
export class CompositeSummaryDownloadComponent implements OnDestroy, OnInit {

  errorText: string;
  file$: Observable<FileModel>;
  id: string;
  paramsSubscription: Subscription;

  constructor(private route: ActivatedRoute, private store: Store<fromFileReducer.State>) {

  }

  ngOnDestroy() {
    if (this.paramsSubscription && !this.paramsSubscription.closed) {
      this.paramsSubscription.unsubscribe();
    }
  }

  ngOnInit() {
    this.paramsSubscription = this.route.queryParamMap.subscribe({
      next: this.startDownload,
    });
  }

  startDownload = (params: ParamMap) => {
    this.id = params.get('compositeDataLoadExternalId');

    if (!RegexStrings.GUID.test(this.id)) {
      this.errorText = `Invalid composite data load external ID: ${this.id}`;
    } else {
      this.store.dispatch(new CompositeSummaryDownload({
        Id: this.id,
        FileType: FileType.InvalidRecordsFile
      }));

      this.file$ = this.store.pipe(
        select(fromFileReducer.selectFileById, this.id),
      );
    }
  }

}
