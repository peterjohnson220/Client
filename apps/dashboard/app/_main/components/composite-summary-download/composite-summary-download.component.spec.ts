import { NO_ERRORS_SCHEMA } from '@angular/core';
import { ComponentFixture, TestBed } from '@angular/core/testing';
import { ActivatedRoute, convertToParamMap, ParamMap } from '@angular/router';
import { Store, StoreModule } from '@ngrx/store';
import { BehaviorSubject } from 'rxjs';

import {
  FileDownloadError,
  FileDownloadProgress,
  FileDownloadSuccess,
} from 'libs/features/file-download/actions/file-download.actions';
import * as fromFileReducer from 'libs/features/file-download/reducers/file-download.reducer';
import { FileModel } from 'libs/models/file';
import * as fromRootState from 'libs/state/state';
import { CompositeSummaryDownload } from '../../actions/composite-summary-download.actions';
import { CompositeSummaryDownloadComponent } from './composite-summary-download.component';

describe('Dashboard - Main - Composite Summary Download', () => {
  let fixture: ComponentFixture<CompositeSummaryDownloadComponent>;
  let instance: CompositeSummaryDownloadComponent;
  let store: Store<fromRootState.State>;
  const goodIdQueryParams = { compositeDataLoadExternalId: 'a9a93799-3825-5804-b6d7-ad87ebee956e' };
  const badIdQueryParams = { compositeDataLoadExternalId: 'invalidId' };

  let queryParamMap: BehaviorSubject<ParamMap>;

  beforeEach(() => {
    queryParamMap = new BehaviorSubject<ParamMap>(convertToParamMap(goodIdQueryParams));

    TestBed.configureTestingModule({
      imports: [
        StoreModule.forRoot({}),
        StoreModule.forFeature('feature_fileDownload', fromFileReducer.reducer),
      ],
      declarations: [
        CompositeSummaryDownloadComponent,
      ],
      providers: [
        {
          provide: ActivatedRoute,
          useValue: {
            queryParamMap,
          },
        },
      ],
      schemas: [ NO_ERRORS_SCHEMA ]
    });

    store = TestBed.get(Store);
    jest.spyOn(store, 'dispatch');

    fixture = TestBed.createComponent(CompositeSummaryDownloadComponent);
    instance = fixture.componentInstance;
  });

  it('should dispatch a CompositeSummaryDownload action on init', () => {
    // arrange
    const action = new CompositeSummaryDownload({
      Id: goodIdQueryParams.compositeDataLoadExternalId,
    });

    // act
    fixture.detectChanges();

    // assert
    expect(store.dispatch).toHaveBeenLastCalledWith(action);
  });

  it('should show an error with an invalid compositeDataLoadExternalId', () => {
    // arrange
    queryParamMap.next(convertToParamMap(badIdQueryParams));

    // act
    fixture.detectChanges();

    // assert
    expect(instance.errorText).toEqual(`Invalid composite data load external ID: ${badIdQueryParams.compositeDataLoadExternalId}`);
  });

  it('should match query parameter error snapshot', () => {
    // arrange
    queryParamMap.next(convertToParamMap(badIdQueryParams));

    // act
    fixture.detectChanges();

    // assert
    expect(fixture).toMatchSnapshot();
  });

  it('should match file error snapshot', () => {
    // arrange
    const fileId = goodIdQueryParams.compositeDataLoadExternalId;
    const file: FileModel = {
      error: true,
      errorText: 'There is an error',
      fileId,
      progress: 100,
    };
    const action = new FileDownloadError(file);
    store.dispatch(action);

    // act
    fixture.detectChanges();

    // assert
    expect(fixture).toMatchSnapshot();
  });

  it('should match file download in progress snapshot', () => {
    // arrange
    const fileId = goodIdQueryParams.compositeDataLoadExternalId;
    const file: FileModel = {
      error: false,
      fileId,
      progress: 50,
    };
    const action = new FileDownloadProgress(file);
    store.dispatch(action);

    // act
    fixture.detectChanges();

    // assert
    expect(fixture).toMatchSnapshot();
  });

  it('should match file download success snapshot', () => {
    // arrange
    const fileId = goodIdQueryParams.compositeDataLoadExternalId;
    const file: FileModel = {
      error: false,
      fileId,
      progress: 100,
    };
    const action = new FileDownloadSuccess(file);
    store.dispatch(action);

    // act
    fixture.detectChanges();

    // assert
    expect(fixture).toMatchSnapshot();
  });
});
