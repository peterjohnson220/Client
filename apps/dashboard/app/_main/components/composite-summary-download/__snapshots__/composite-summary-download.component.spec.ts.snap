// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard - Main - Composite Summary Download should match file download in progress snapshot 1`] = `
<pf-composite-summary-download
  file$={[Function Store]}
  id={[Function String]}
  paramsSubscription={[Function Subscriber]}
  route={[Function Object]}
  startDownload={[Function Function]}
  store={[Function Store]}
>
  <div
    class="w-75 mx-auto mt-4 text-white text-center"
  >
    <h1>
      Downloading file...
    </h1>
    
    <div>
      
      <h2>
        Please stand by.
      </h2>
      
      <ngb-progressbar
        class="d-block mt-4"
        type="success"
      />
    </div>
    
  </div>
</pf-composite-summary-download>
`;

exports[`Dashboard - Main - Composite Summary Download should match file download success snapshot 1`] = `
<pf-composite-summary-download
  file$={[Function Store]}
  id={[Function String]}
  paramsSubscription={[Function Subscriber]}
  route={[Function Object]}
  startDownload={[Function Function]}
  store={[Function Store]}
>
  <div
    class="w-75 mx-auto mt-4 text-white text-center"
  >
    <h1>
      Downloading file...
    </h1>
    
    <div>
      
      <h2>
        Please stand by.
      </h2>
      
      <ngb-progressbar
        class="d-block mt-4"
        type="success"
      />
    </div>
    
  </div>
</pf-composite-summary-download>
`;

exports[`Dashboard - Main - Composite Summary Download should match file error snapshot 1`] = `
<pf-composite-summary-download
  file$={[Function Store]}
  id={[Function String]}
  paramsSubscription={[Function Subscriber]}
  route={[Function Object]}
  startDownload={[Function Function]}
  store={[Function Store]}
>
  <div
    class="w-75 mx-auto mt-4 text-white text-center"
  >
    <h1>
      Downloading file...
    </h1>
    
    <div>
      
      <h2
        class="text-danger"
      >
        Error: There is an error
      </h2>
      
    </div>
    
  </div>
</pf-composite-summary-download>
`;

exports[`Dashboard - Main - Composite Summary Download should match query parameter error snapshot 1`] = `
<pf-composite-summary-download
  errorText={[Function String]}
  id={[Function String]}
  paramsSubscription={[Function Subscriber]}
  route={[Function Object]}
  startDownload={[Function Function]}
  store={[Function Store]}
>
  <div
    class="w-75 mx-auto mt-4 text-white text-center"
  >
    <h1>
      Downloading file...
    </h1>
    
    <div>
      
      
      <h2
        class="text-danger"
      >
        Error: Invalid composite data load external ID: invalidId
      </h2>
    </div>
    
  </div>
</pf-composite-summary-download>
`;
